src/main/scala/jigg/util/XMLUtil.scala

```diff
@@ -119,13 +119,13 @@ object XMLUtil {
   def findAllSub(node: Node, that: String): java.util.List[Node] = node \\ that
 
   def hasChild(nodes: NodeSeq): Boolean = nodes match {
-    case x: Elem => x.child.length > 0
+    case x: Elem => getNonEmptyChild(x).length > 0
     case _ => nodes forall hasChild
   }
 
-  def getChildNode(nodes: NodeSeq): NodeSeq = nodes match {
-    case x: Elem => x.child filter (_.label != "#PCDATA")
-    case x => (x.map(getChildNode(_))).flatten
+  def getNonEmptyChild(nodes: NodeSeq): NodeSeq = nodes match {
```

Comment:
This and `hasChild` method may be very inefficient when the size of XML tree is huge. Is it possible to avoid traversing all nodes in the tree?